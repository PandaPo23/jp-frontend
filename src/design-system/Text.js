import Box from './Box';
import system from '../utils/System';

const Text = system(
  'Text',
  {
    extend: Box,
  },
  ({ center, right }) => ({
    textAlign: (center || right) && (center ? 'center' : 'right'),
  }),
  ({ bold }) => ({
    fontWeight: bold && 'bold',
  }),
  ({ uppercase, lowercase, capitalize, nobr, truncate }) => ({
    textTransform: uppercase
      ? 'uppercase'
      : lowercase
      ? 'lowercase'
      : capitalize
      ? 'capitalize'
      : null,
    whiteSpace: truncate || nobr ? 'nowrap' : undefined,
    textOverflow: truncate ? 'ellipsis' : undefined,
    overflow: truncate ? 'hidden' : undefined,
  })
);

export default Text;
