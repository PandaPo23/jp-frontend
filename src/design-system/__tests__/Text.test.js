import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Text from '../Text';

describe('Text component', () => {
  it('renders uppercase Text correctly', () => {
    const localProps = {
      uppercase: true,
      children: 'Upper Case',
    };
    const wrapper = mount(<Text {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders lowercase Text correctly', () => {
    const localProps = {
      textStyle: 'lowercase',
      children: 'Lower Case',
    };
    const wrapper = mount(<Text {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders capitalized Text correctly', () => {
    const localProps = {
      textStyle: 'capitalize',
      children: 'Capitalized',
    };
    const wrapper = mount(<Text {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders truncated Text correctly', () => {
    const localProps = {
      textStyle: 'truncate',
      children: 'Truncated',
    };
    const wrapper = mount(<Text {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
