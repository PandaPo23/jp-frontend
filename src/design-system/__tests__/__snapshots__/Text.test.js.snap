// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component renders capitalized Text correctly 1`] = `
<Text
  textStyle="capitalize"
>
  <div
    className="emotion-0"
  >
    Capitalized
  </div>
</Text>
`;

exports[`Text component renders lowercase Text correctly 1`] = `
<Text
  textStyle="lowercase"
>
  <div
    className="emotion-0"
  >
    Lower Case
  </div>
</Text>
`;

exports[`Text component renders truncated Text correctly 1`] = `
<Text
  textStyle="truncate"
>
  <div
    className="emotion-0"
  >
    Truncated
  </div>
</Text>
`;

exports[`Text component renders uppercase Text correctly 1`] = `
.emotion-0 {
  text-transform: uppercase;
}

<Text
  uppercase={true}
>
  <div
    className="emotion-0"
  >
    Upper Case
  </div>
</Text>
`;
